### PR TITLE
Login ui

### PR DESCRIPTION
**Pull request by bilalb**

### CODE REVIEW

File: Subsonicious/Subsonicious/Views/LoginView.swift [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+//  Created by Bilal on 15/06/2020.
+//  Copyright © 2020 Bilal Benlarbi. All rights reserved.
+//
+
+import SwiftUI
+
+struct LoginView: View {
+
+    @EnvironmentObject var authentication: Authentication
+    @State var server = Server()
```
Comment (zarghol) [inline Subsonicious/Subsonicious/Views/LoginView.swift:14]: state = private please :P

Comment (zarghol) [inline Subsonicious/Subsonicious/Views/LoginView.swift:14]: state = private please :P

File: Subsonicious/Subsonicious/Views/LoginView.swift [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    var body: some View {
+        GeometryReader { content in
+            VStack {
+                Text("login.login")
+
+                VStack {
+                    TextField("login.server", text: self.$server.address)
+                        .keyboardType(.URL)
+                        .textContentType(.URL)
```
Comment (zarghol) [inline Subsonicious/Subsonicious/Views/LoginView.swift:24]: I think the contentType changes the keyboard type too, so you don't need the first line :)

File: Subsonicious/Subsonicious/Views/LoginView.swift [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                        .asLoginField()
+
+                    TextField("login.username", text: self.$server.username)
+                        .keyboardType(.asciiCapable)
```
Comment (zarghol) [inline Subsonicious/Subsonicious/Views/LoginView.swift:28]: same :)

File: Subsonicious/Subsonicious/Views/LoginView.swift [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                        .textContentType(.username)
+                        .asLoginField()
+
+                    SecureField("login.password", text: self.$server.password)
+                        .textContentType(.password)
+                        .asLoginField()
+                }
+
+                Button(action: {
```
Comment (zarghol) [inline Subsonicious/Subsonicious/Views/LoginView.swift:37]:
```suggestion
                Button(action: continueButtonPressed)
```

Comment (zarghol) [reply Subsonicious/Subsonicious/Views/LoginView.swift:37]: I don't know to add multiple line suggestion on GitHub ^^'

File: Subsonicious/Subsonicious/Views/LoginView.swift [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+                    self.continueButtonPressed()
+                }) {
+                    Text("login.continue")
+                        .font(.headline)
+                        .foregroundColor(.white)
+                        .frame(maxWidth: content.size.width, maxHeight: 44)
```
Comment (zarghol) [inline Subsonicious/Subsonicious/Views/LoginView.swift:43]: why do you need to specify maxWidth like that ? could it be done with vertical padding instead ?

File: Subsonicious/Subsonicious/Views/LoginView.swift [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+                        .background(Color.yellow)
+                        .cornerRadius(6)
+                        .opacity(self.continueButtonOpacity)
+                }
+                .disabled(self.shouldDisableContinueButton)
```
Comment (zarghol) [inline Subsonicious/Subsonicious/Views/LoginView.swift:48]: I think you can rename the variable to `continueButtonIsDisabled` to symbolise the current state, not a futur with should.

File: Subsonicious/Subsonicious/Views/LoginView.swift [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    }
+}
+
+struct LoginView_Previews: PreviewProvider {
+    static var previews: some View {
+        LoginView()
+    }
+}
+
+struct Server {
```
Comment (zarghol) [inline Subsonicious/Subsonicious/Views/LoginView.swift:80]: move to different files the model and extension :)

File: Subsonicious/Subsonicious/Views/LoginView.swift (hunk 2 above)
Comment (bilalb) [reply Subsonicious/Subsonicious/Views/LoginView.swift:24]: No, it doesn't

File: Subsonicious/Subsonicious/Views/LoginView.swift (hunk 5 above)
Comment (bilalb) [reply Subsonicious/Subsonicious/Views/LoginView.swift:43]: In order for the text to take the whole width available :)
Didn't manage to achieve that with padding

**Pull request merged**